miniapps/gslib/ex1.cpp

```diff
@@ -0,0 +1,388 @@
+   {
+      for (int i = 0; i < nmeshes; i ++)
+      {
+         b_ar[i] = new LinearForm(fespacearr[i]);
+         b_ar[i]->AddDomainIntegrator(new DomainLFIntegrator(one));
+         b_ar[i]->Assemble();
+      }
+
+      OperatorPtr A1, A2;
+      Vector B1, X1, B2, X2;
+
+      a_ar[0]->FormLinearSystem(ess_tdof_list1, x1, *b_ar[0], A1, X1, B1);
+      a_ar[1]->FormLinearSystem(ess_tdof_list2, x2, *b_ar[1], A2, X2, B2);
+
+      // Solve the linear system A X = B.
+      // Use a simple symmetric Gauss-Seidel preconditioner with PCG.
+      GSSmoother M1((SparseMatrix&)(*A1));
+      PCG(*A1, M1, B1, X1, 0, 200, 1e-12, 0.0);
+      GSSmoother M2((SparseMatrix&)(*A2));
+      PCG(*A2, M2, B2, X2, 0, 200, 1e-12, 0.0);
+
+      // Recover the solution as a finite element grid function.
+      a_ar[0]->RecoverFEMSolution(X1, *b_ar[0], x1);
+      a_ar[1]->RecoverFEMSolution(X2, *b_ar[1], x2);
+
+      // Interpolate boundary condition
+      finder1.Interpolate(x1, interp_vals2);
+      finder2.Interpolate(x2, interp_vals1);
+
+      double dxmax = std::numeric_limits<float>::min();
+      double x1inf = x1.Normlinf();
+      double x2inf = x2.Normlinf();
+      for (int i = 0; i < nb1; i++)
+      {
+         int idx = ess_tdof_list1_int[i];
+         double dx = std::abs(x1(idx)-interp_vals1(i))/x1inf;
+         if (dx > dxmax) { dxmax = dx; }
+         x1(idx) = interp_vals1(i);
+      }
+
+      for (int i = 0; i < nb2; i++)
+      {
+         int idx = ess_tdof_list2_int[i];
+         double dx = std::abs(x2(idx)-interp_vals2(i))/x2inf;
+         if (dx > dxmax) { dxmax = dx; }
+         x2(idx) = interp_vals2(i);
+      }
+
+      delete b_ar[0];
+      delete b_ar[1];
+
+      std::cout << std::setprecision(8)    <<
+                "Iteration: "           << schwarz <<
+                ", Relative residual: " << dxmax   << endl;
+      if (dxmax < rel_tol) { break; }
+   }
+
+   // Send the solution by socket to a GLVis server.
+   {
+      char vishost[] = "localhost";
+      int  visport   = 19916;
+      for (int ip = 0; ip<mesharr.Size(); ++ip)
+      {
+         socketstream sol_sock(vishost, visport);
+         sol_sock.precision(8);
+         sol_sock << "parallel " << mesharr.Size() << " " << ip << "\n";
+         if (ip==0) { sol_sock << "solution\n" << *mesharr[ip] << x1 << flush; }
+         if (ip==1) { sol_sock << "solution\n" << *mesharr[ip] << x2 << flush; }
+         sol_sock << "window_title 'Overlapping grid solution'\n"
+                  << "window_geometry "
+                  << 0 << " " << 0 << " " << 450 << " " << 350 << "\n"
+                  << "keys jmcA]]]" << endl;
+      }
+   }
+
+   // Free the used memory.
+   finder1.FreeData();
+   finder2.FreeData();
+   for (int i = 0; i < nmeshes; i++)
+   {
+      delete a_ar[i];
+      delete fespacearr[i];
+      if (order > 0) { delete fecarr[i]; }
+      delete mesharr[i];
+   }
+
+   return 0;
+}
+
+void GetInterdomainBoundaryPoints(FindPointsGSLIB &finder1,
```

Comment:
I agree Jamie.. I thought about it and decided to keep it here for now because as we learn about more use cases in future, we might have to break backwards compatibility to accommodate them if we put this in source right now. We have already had to that once for `gslib`.